docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -63,119 +66,79 @@ audience. For instance:
 
-A human user completes the OAuth flow through a web browser, obtaining an access token
-that identifies them personally and allows the client to act on their behalf.
+The authorization server is beyond the scope of this specification. It may be the same server as the
+resource server or a separate entity. Section [2.3 Authorization Server Discovery](#2-3-authorizaton-server-discovery)
+specifies how an MCP server (OAuth 2.1 resource server) indicates the location of an OAuth 2.1 authorization server to an OAuth 2.1 client.
 
-When authorization is required and not yet proven by the client, servers **MUST** respond
-with _HTTP 401 Unauthorized_.
+### 2.3 Authorization Server and Capability Discovery
+This section details how MCP servers can inform MCP clients of authorization servers, and how MCP clients discover authorization servers and their capabilities.
 
-Clients initiate the
-[OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12#name-authorization-code-grant)
-authorization flow after receiving the _HTTP 401 Unauthorized_.
+### 2.3.1 Authorization Server Location
+MCP servers **MUST** implement the [OAuth 2.0 Protected Resource Metadata](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-resource-metadata-13)
+specification to indicate the locations of authorization servers. The Protected Resource Metadata document returned by the MCP server **MUST** include
+the `authorization_servers` field.
 
-The following demonstrates the basic OAuth 2.1 for public clients using PKCE.
+The specific use of `authorization_servers` is beyond the scope of this specification; implementers should consult
+the [OAuth 2.0 Protected Resource Metadata](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-resource-metadata-13) documentation for
+comprehensive guidance on implementation details.
 
-```mermaid
-sequenceDiagram
-    participant B as User-Agent (Browser)
-    participant C as Client
-    participant M as MCP Server
+MCP servers **MAY** use the HTTP header `WWW-Authenticate` when returning a _401 Unauthorized_ to indicate the location of the authorization
+server as described in [OAuth 2.0 Protected Resource Metadata](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-resource-metadata-13).
 
-    C->>M: MCP Request
-    M->>C: HTTP 401 Unauthorized
-    Note over C: Generate code_verifier and code_challenge
-    C->>B: Open browser with authorization URL + code_challenge
-    B->>M: GET /authorize
-    Note over M: User logs in and authorizes
-    M->>B: Redirect to callback URL with auth code
-    B->>C: Callback with authorization code
-    C->>M: Token Request with code + code_verifier
-    M->>C: Access Token (+ Refresh Token)
-    C->>M: MCP Request with Access Token
-    Note over C,M: Begin standard MCP message exchange
-```
+MCP clients **MUST** handle the `WWW-Authenticate` header if present, and otherwise use direct discovery via the well-known URI if `WWW-Authenticate`.
```

Comment:
```suggestion
MCP clients **MUST** handle the `WWW-Authenticate` header if present, and otherwise use direct discovery via the well-known URI if no `WWW-Authenticate` header is present.
```